src/controllers/authController.ts

```diff
@@ -0,0 +1,75 @@
+import { Request, Response } from "express";
+import User from "../models/userModel";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import { sendConfirmEMail } from "../config/nodemailerConfig";
+import { generateCode } from "../config/generateCode";
+import * as dotenv from 'dotenv'
+
+dotenv.config();
+
+export const createUser = async (req: Request, res: Response) => {
+    const code = generateCode();
+
+    const newUser = new User({
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        email: req.body.email,
+        mobile: req.body.mobile,
+        password: bcrypt.hashSync(req.body.password, 8),
+        active: false,
+        verificationCode: code,
+    });
+
+    try {
+        const emailCheck = await User.findOne({
+            $or: [{ email: newUser.email }],
+        });
+        if (emailCheck) {
+            return res.status(400).send({
+                message:
+                    "This Email is already in use, please login to your account",
+            });
+        }
+
+        newUser
+            .save()
+            .then((user) => {
+                sendConfirmEMail(
+                    user.firstName,
+                    user.email,
+                    user.verificationCode
+                );
+                const message = "Check your email for confirmation!";
+                res.status(201).json({ message, user });
+            })
+            .catch((error) => {
+                res.status(404).json(error);
+            });
+    } catch (error) {
+        res.status(400).send(error);
+    }
+};
+
+export const verifyUserEmail = (req: Request, res: Response) => {
```

Comment:
Please refactor this to use async await and abstract all business logic and import them for use when needed